src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -732,7 +732,7 @@ def show_application(client, identifier):
 
 def delete_application(client, identifier):
     object_id = _resolve_application(client, identifier)
-    client.application_delete(id=object_id)
+    client.application_delete(object_id)
```

Comment:
If the function takes positional argument, the best practice is to pass the argument as positional argument too, instead of as a keyword argument.